### PR TITLE
Adding Contributing.md file

### LINKED ISSUE #100: Create System.Device.Gpio Overview doc
It would be beneficial to create a **System.Device.Gpio Overview** markdown to help newbies get started along with more technical details on infrastructure.  It could include the following topics:

**NOTE**: The intent of this overview is to focus on the following:
1. Intro examples of creating a protocol connection (e.g. I2C, SPI) and instantiating the respective device (e.g. I2cDevice, SpiDevice).  This would be more focused on how it relates to interface topics (e.g. buses, addressing, etc.);  not so much on particular binding.
2. Understanding how the OS and/or specific device is determined.
3. Understanding of external bindings.
4. Other topics?

⚠️  It might be more appropriate to document certain topics in [Microsoft Docs](https://docs.microsoft.com) to keep consistent with the overall .NET ecosystem.

- [ ] Create/add diagram helper showing the structure along with sections for future work.
- [ ] Add System.Device.Gpio API folder and placeholder markdown files under [Documentation](https://github.com/dotnet/iot/tree/master/Documentation).
- [ ] Update [Documentation](https://github.com/dotnet/iot/tree/master/Documentation) with other types of interfaces.
- [ ] Update [Documentation System.Device.* APIs section](https://github.com/dotnet/iot/tree/master/Documentation#systemdevice-apis) with link to System.Device.Gpio API README markdown.
- [ ] Add link to System.Device.Gpio API README in [main repo README markdown](https://github.com/dotnet/iot/blob/master/README.md).

### PR DESCRIPTION
**Pull request by joperezr**
Fixes #100 

cc: @shaggygi @krwq @richlander 

This PR will finally add a Contributing.md file which should provide the right guidance in order to start contributing to our repo. Please let me know if I missed anything obvious and I'll be sure to add it here.

### CODE REVIEW

File: Documentation/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+Building IoT on Windows
```
Comment (krwq) [inline Documentation/CONTRIBUTING.md]: Would it make sense to put windows specific instructions to something like:
Contributing-Windows.md and then point there from here?

I think this should contain OS-independent instructions

File: Documentation/README.md [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
+1. **IoT.Device.Bindings** device bindings library. These is a collection of types which work as wrappers (or bindings) for devices and sensors which are able to talk to a microcontroller unit (or MCU like a Raspberry Pi for example) using the protocols supported by System.Device.Gpio. For example: [BME280](/../src/devices/Bmxx80/README.md) is a temperature sensor which uses SPI and I2C in order to communicate with a MCU and is able to report the current temperature. Because the process of how to compute the temperature from the data is not trivial, we have a `Bme280` class which exposes friendly methods like `ReadTemperature()` which will internally use either SPI or I2C to get the current temperature value. In order to start adding a new binding, check out our [guide on how to contribute a new binding](/../tools/templates/DeviceBindingTemplate/README.md).
```
Comment (krwq) [inline Documentation/README.md]: might be worth to mention that src/Iot.Device.Bindings contains the project but src/devices is the actual implementation

File: Documentation/CONTRIBUTING.md (hunk 1 above)
Comment (joperezr) [reply Documentation/CONTRIBUTING.md]: you are right, in fact most of these instructions are os-independent. I will fix this in the next commit.

**Comment by joperezr** [comment]
I'm merging this in now to close the documentation issue. I will address any subsequent feedback in a new PR.

**Pull request merged**